lib/engine/round/g_1846/operating.rb

```diff
@@ -43,6 +43,12 @@ def after_process(action)
             end
           end
 
+          dividend_step = steps.find { |step| step.class == Step::G1846::Dividend }
```

Comment:
maybe this should be

unless current_step.is_a?(Steps::1846::Dividend)
steps.find { ... is_a? }.pass!

Reply:
The dividend step doesn't happen if no trains run. Will after_process() still be called in that case? When does current_step change to the next step?

Reply:
Confirmed the dividend step can be non-blocking, so it doesn't always become the current_step.

Reply:
ok then let's just do

unless steps.find { |step| step.is_a?(Step::) }.active?

Reply:
Fixed.